flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -779,6 +779,19 @@ JobManagerRunner createJobManagerRunner(JobGraph jobGraph, long initializationTi
                                 operatorId, serializedRequest, timeout));
     }
 
+    @Override
+    public void submitFailedJob(FatalProgramInvocationException exception) {
```

Comment:
Why do we need this method? We didn't really execute the job, so there is nothing to archive.

Reply:
The idea is that the user can receive the application result so we have to submit something containing the error message.